shared/src/androidMain/kotlin/io/customer/shared/database/DatabaseDriverFactory.kt

```diff
@@ -3,10 +3,19 @@ package io.customer.shared.database
 import android.content.Context
 import com.squareup.sqldelight.android.AndroidSqliteDriver
 import com.squareup.sqldelight.db.SqlDriver
+import io.customer.shared.AndroidPlatform
+import io.customer.shared.Platform
 import io.customer.shared.local.CioDatabase
 
 actual class DatabaseDriverFactory(private val context: Context) {
-    actual fun createDriver(): SqlDriver {
-        return AndroidSqliteDriver(CioDatabase.Schema, context, "customerio.db")
-    }
+    actual fun createDriver(databaseName: String): SqlDriver = AndroidSqliteDriver(
+        schema = CioDatabase.Schema,
+        context = context,
+        name = databaseName,
+    )
 }
+
+// Platform in Android classes will always be [AndroidPlatform]
+actual fun getDatabaseDriverFactory(platform: Platform) = DatabaseDriverFactory(
+    context = (platform as AndroidPlatform).applicationContext
```

Comment:
Instead of doing a cast, I think you can create an `expect typealias`. 

In commonMain, create `expect typealias Platform`. 
In androidMain, `actual typealias Platform = AndroidPlatform`. 

Then, anywhere you use `Platform` in androidMain, you don't need to cast it.

Reply:
Yeah, this looks a good idea, will add this

Reply:
Looks like `expect typealias` directly isn't allowed by KMM but I have followed similar approach to achieve this and pushed the changes. We don't need casting it anywhere now.